dashboard/test/controllers/redirect_proxy_controller_test.rb

```diff
@@ -0,0 +1,58 @@
+require 'webmock/minitest'
+WebMock.disable_net_connect!(:allow_localhost => true)
+require_relative '../../../shared/test/spy_newrelic_agent'
+require 'test_helper'
+
+class RedirectProxyControllerTest < ActionController::TestCase
+  LONG_URI = 'https://studio.code.org/s/course1/stage/3/puzzle/2'
+
+  test "should successfully get a redirect for url in whitelist" do
+    short_uri = 'http://bit.ly/2gPJmTQ'
```

Comment:
TIL that bit.ly links never expire (as long as the service remains up) http://webapps.stackexchange.com/questions/10987/do-customized-bit-ly-links-ever-expire.